rust/agama-server/src/network/model.rs

```diff
@@ -1105,6 +1186,108 @@ impl TryFrom<&str> for SecurityProtocol {
     }
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Serialize)]
+pub enum GroupAlgorithm {
+    Wep40,
+    Wep104,
+    Tkip,
+    Ccmp,
+}
+
+#[derive(Debug, Error)]
+#[error("Invalid group algorithm: {0}")]
+pub struct InvalidGroupAlgorithm(String);
+
+impl FromStr for GroupAlgorithm {
```

Comment:
For some time, I have been thinking whether we should use [strum](https://docs.rs/strum/latest/strum/) (see the [EnumString](https://docs.rs/strum/latest/strum/derive.EnumString.html) example) to avoid all those trivial conversions.

I am not sure about how error handle would be, but I think it is worth having a closer look.

In any case, it is out of the scope of this PR.

Reply:
Totally agree, strum would make this a lot simpler :slightly_smiling_face: